Microsoft.Azure.Cosmos.Encryption/src/EncryptionContainer.cs

```diff
@@ -45,6 +56,10 @@ internal sealed class EncryptionContainer : Container
 
         public override Database Database => this.container.Database;
 
+        public Dictionary<List<string>, string> ToEncrypt = new Dictionary<List<string>, string>();
```

Comment:
Does this  need to be public? Should a new object be created to explain this structure. It's difficult to understand what this structure actually contains.

```suggestion
        public IReadOnlyDictionary<List<string>, string> ToEncrypt {get;}
```

Reply:
This is a Dictionary of paths and their corresponding key. It is used to pass around the properties the user chooses to encrypt with their keys. I will add a new object to make it clearer and also mark it as internal .